source/components/Autocomplete.js

```diff
@@ -305,7 +313,16 @@ class AutocompleteBase extends Component<AutocompleteProps, State> {
             error={error || this.state.error}
             filteredOptions={this.state.filteredOptions}
             getSelectionProps={this.getSelectionProps}
-            handleAutocompleteClick={this.handleAutocompleteClick}
+            handleAutocompleteClick={() => {
+              setTimeout(() => {
+                const { isRemoveWordClicked } = this.state;
+                if (!isRemoveWordClicked) {
+                  this.handleAutocompleteClick();
+                } else {
+                  this.setState((prevState) => ({ isRemoveWordClicked: !prevState.isRemoveWordClicked }));
+                }
+              }, 0);
+            }}
```

Comment:
@aleksandardjordjeviciohk can we improve the `isRemoveWordClicked` logic to something cleaner? It looks like a hack to me and in general I would like to keep timeouts to the absolute minimum because they are also hacks in most cases.